### PR TITLE
Add intersection support.

### PR DESCRIPTION
**Pull request by minism**
Related change: https://github.com/gristlabs/ts-interface-builder/pull/20

Error messages have not been changed beyond the default yet, though the defaults from the library are already very useful, indicating which keys are missing.

Please let me know thoughts on the overall design approach here, if any major changes are needed, etc.  Thanks!

### CODE REVIEW

File: test/test_checker.ts [hunk 1]
```diff
@@ -6,6 +6,8 @@ import sample from "./fixtures/sample-ti";
 import shapes from "./fixtures/shapes-ti";
 import * as enumUnion from "./fixtures/enum-union";
 import enumUnionTI from "./fixtures/enum-union-ti";
+import * as intersection from "./fixtures/intersection";
```
Comment (dsagal) [inline test/test_checker.ts:9]: This import is unused as far as I can tell.

File: test/test_checker.ts [hunk 2]
```diff
@@ -239,6 +241,21 @@ describe("ts-interface-checker", () => {
...
+    Car.check({ numDoors: 2, numWheels: 4})
+    House.check({ numDoors: 6, numRooms: 8 })
+
+    assert.throws(() => Car.check({ numDoors: 2 }),
+      /value.numWheels is missing/);
+    assert.throws(() => Car.check({ numWheels: 4 }),
+      /value.numDoors is missing/);
+    assert.throws(() => House.check({ numDoors: 6 }),
+      /value.numRooms is missing/);
+    assert.throws(() => House.check({ numRooms: 8 }),
```
Comment (dsagal) [inline test/test_checker.ts:255]: I think some variety of tests would be useful here: a type with more than 2 parts of intersection, a check that fails more than one of them, a check that fails in a different way (e.g. `{numDoors: "foo"}`. Also a strictCheck() (but see earlier comment).

File: test/test_checker.ts (hunk 1 above)
Comment (minism) [reply test/test_checker.ts:9]: Removed.

File: test/test_checker.ts (hunk 2 above)
Comment (minism) [reply test/test_checker.ts:254]: Okay, i believe I was able to implement strict check correctly, as well as added some more test cases from your suggestions.

**Comment by dsagal** [comment]
I poked around and found a slightly simpler approach to checking extra properties, made a PR on top of your branch here: https://github.com/akoralabs/ts-interface-checker/pull/1

What do you think?

**Comment by minism** [comment]
> I poked around and found a slightly simpler approach to checking extra properties, made a PR on top of your branch here: [akoralabs#1](https://github.com/akoralabs/ts-interface-checker/pull/1)
> 
> What do you think?

Looks awesome, thank you!  Just merged it.

**Pull request merged**